backend/fms_core/filters.py

```diff
@@ -75,6 +76,23 @@ def is_pooled_filter(self, queryset, name, values):
         bool_value = (values == 'true')
         return queryset.filter(is_pooled=bool_value)
 
+    def metadata_filter(self, queryset, name, values):
+        condition = Q()
+        # Metadata is of form: 'name1__value1, name2__value2, name2__Value3'
+        for metadatum in values.split(','):
+            if metadatum:
+                name_value = metadatum.split('__')
+                # Check that there's at least 1 '__' that separates name and value
+                if name_value and len(name_value) > 1:
+                    # Meaning value was left empty
+                    if not name_value[1]:
+                        condition |= Q(name=name_value[0])
+                    # Use the string after the first '__' as the value (even if it contained '__')
+                    else:
+                        condition |= Q(name=name_value[0], value='__'.join(name_value[1:]))
```

Comment:
you can use the maxsplit=1 param of split() to save the need to join after.

Reply:
Great suggestion. Didn't know about this capability.